extensions/azurecore/src/azureResource/services/subscriptionService.ts

```diff
@@ -50,7 +51,7 @@ export class AzureResourceSubscriptionService implements IAzureResourceSubscript
 					void vscode.window.showWarningMessage(errorMsg);
 				}
 			} catch (error) {
-				if (!account.isStale) {
+				if (!account.isStale && !error.message.includes(TenantIgnoredErrorCode)) {
```

Comment:
> Or - what about just skipping the tenant in the for loop above if it's in the ignore list?

That would mean I'd have to return undefined, and that can be difficult to track when to expect 'undefined' and when not.

Reply:
Why would you have to return undefined? If there aren't any subscriptions (or they're all in ignored tenants) I'd expect it to just return an empty array.

Reply:
Currently ADAL returns undefined for stale accounts and other cases - so to avoid mixing it with stale accounts.. and triggering reauthentication, we should throw error. The caller code can handle this accordingly I think, since they're sending tenant Id.

Reply:
Not always though - the tenantIds parameter is optional. And the only time we use that parameter is when we're iterating over all the tenants of the account anyways from what I see. So it does look like my guess about there being other places that we're iterating over all tenants on a subscription is true, which is something that should also really be fixed to avoid unnecessary overhead and potential errors.